lib/ecto/adapters/postgres/connection.ex

```diff
@@ -682,6 +682,13 @@ if Code.ensure_loaded?(Postgrex) do
       "ALTER TABLE #{quote_table(constraint.prefix, constraint.table)} DROP CONSTRAINT #{quote_name(constraint.name)}"
     end
 
+    def execute_ddl({:create, %View{} = view}) do
+      {query, params, _key} = Ecto.Query.Planner.prepare(view.query, :all, Ecto.Adapters.Postgres, 0)
```

Comment:
Great, totally agree. Will try to get `Ecto.Adapters.SQL.to_sql` to work without an application repo, and move this piece of logic to the `Ecto.Migration`